src/renderer/components/deposit/withdraw/Withdraw.tsx

```diff
@@ -105,26 +322,43 @@ export const Withdraw: React.FC<Props> = ({
             trimZeros: true
           })}
           {/* show pricing if price asset is different only */}
-          {!eqAsset.equals(depositAsset, selectedCurrencyAsset) &&
+          {!eqAsset.equals(depositAsset, selectedPriceAsset) &&
             ` (${formatAssetAmountCurrency({
               amount: assetAmount(withdrawAmounts.assetWithdraw.amount().times(assetPrice)),
-              asset: selectedCurrencyAsset,
+              asset: selectedPriceAsset,
               trimZeros: true
             })})`}
         </Styled.OutputLabel>
       </Styled.AssetContainer>
 
-      <Label disabled={RD.isPending(fee)}>
-        <Row align="middle">
-          <ReloadButton onClick={updateFees} disabled={RD.isPending(fee)} />
-          {renderFee}
-        </Row>
-      </Label>
+      <Styled.FeesRow gutter={{ lg: 32 }}>
+        <Col>
+          <Styled.FeeRow>
+            <Col>
+              <ReloadButton onClick={updateFees} disabled={RD.isPending(fees)} />
+            </Col>
+            <Col>
+              <Styled.FeeLabel disabled={RD.isPending(fees)} color={RD.isFailure(fees) ? 'error' : 'normal'}>
+                {renderFee}
+              </Styled.FeeLabel>
+            </Col>
+          </Styled.FeeRow>
+          <Styled.FeeErrorRow>
+            <Col>
+              <>
+                {!!isThorMemoFeeError && renderThorMemoFeeError}
+                {!!isThorOutFeeError && renderThorOutFeeError}
+                {!!isAssetChainFeeError && renderAssetChainFeeError}
```

Comment:
with propose checks for render methods we will not any conditions at the render itself

Reply:
That's correct. However, it's just another way for doing same thing.

Reply:
@veado not exactly. with your implementation all of that stuff will be computed anyway but it will not be shown.

Reply:
You are right in case of computation - I will change it ...

Reply:
8c4fad5